torch/distributed/rpc/server_process_global_profiler.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/python3
+
+import torch
+from torch.autograd import ProfilerConfig
+from torch.autograd.profiler import profile
+
+from . import (
+    __disable_server_process_global_profiler,
+    __enable_server_process_global_profiler,
+)
+
+
+# ==================== Functional API ====================
+
+
+def _enable_server_process_global_profiler(config: ProfilerConfig):
+    r"""
+    Turn on server-side process-global profiling.
+    This enables thread-local profiler on all RPC threads running server-side request callbacks.
+
+    Arguments:
+        config (torch.autograd.profiler.ProfilerConfig): This config specifies
+            1) what mode. CPU only or CPU + CUDA.
+            2) Whether to record input shapes.
+
+    Returns:
+        None
+
+    Example::
+        >>> import torch
+        >>> profiler_config = torch.autograd.ProfilerConfig(
+        >>>     torch.autograd.ProfilerState.CPU,  # profiler_kind
+        >>>     False,  # record_input_shapes
+        >>>     False,  # profile_memory
+        >>> )
+        >>> rpc.enable_server_process_global_profiler(profiler_config)
```

Comment:
Is this the anticipated usage? If so, it looks different from the tests, where an RPC to enable it on the server is being sent.  Should the example be updated to send an RPC (or to the expected usage of the API)?

Reply:
Updating the example to be in a RPC setup.